[angle] static build link error on windows

When link to static angle library, many symbols unresolved.

* OS: windows 10
* Triplet: x64-windows-static

To reproduce:
Check [https://github.com/xiaozhuai/vcpkg_demo](https://github.com/xiaozhuai/vcpkg_demo)
Comment out other demos instead of opengles one.
When use `x64-windows` triplet, it's all good.
When use `x64-windows-static` triplet, it fails.

<details> <summary>Click to show log</summary>

```
LINK: command "C:\PROGRA~2\MICROS~2\2022\BUILDT~1\VC\Tools\MSVC\1435~1.322\bin\Hostx64\x64\link.exe /nologo src\opengles_angle\CMakeFiles\vcpkg_demo_opengles_angle.dir\src\main.cpp.obj src\opengles_angle\CMakeFiles\vcpkg_demo_opengles_angle.dir\src\stb.cpp.obj /out:src\opengles_angle\vcpkg_demo_opengles_angle.exe /implib:src\opengles_angle\vcpkg_demo_opengles_angle.lib /pdb:src\opengles_angle\vcpkg_demo_opengles_angle.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subsystem:console vendor\glfw\src\glfw3.lib vendor\imgui\imgui.lib D:\vcpkg\installed\x64-windows-static\lib\libEGL.lib D:\vcpkg\installed\x64-windows-static\lib\libGLESv2.lib vendor\glfw\src\glfw3.lib D:\vcpkg\installed\x64-windows-static\lib\libEGL.lib D:\vcpkg\installed\x64-windows-static\lib\libGLESv2.lib D:\vcpkg\installed\x64-windows-static\lib\ANGLE.lib D:\vcpkg\installed\x64-windows-static\lib\zlib.lib D:\vcpkg\installed\x64-windows-static\lib\ANGLE.lib D:\vcpkg\installed\x64-windows-static\lib\zlib.lib dxguid.lib dxgi.lib d3d9.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:src\opengles_angle\vcpkg_demo_opengles_angle.exe.manifest" failed (exit code 1120) with the following output:
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glActiveTexture referenced in function "void __cdecl draw(struct GLFWwindow *)" (?draw@@YAXPEAUGLFWwindow@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glActiveTexture
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glAttachShader referenced in function "unsigned int __cdecl createProgram(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createProgram@@YAIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@0@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glAttachShader
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glBindTexture referenced in function "unsigned int __cdecl createTextureFromFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createTextureFromFile@@YAIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glBindTexture
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glClear referenced in function "void __cdecl draw(struct GLFWwindow *)" (?draw@@YAXPEAUGLFWwindow@@@Z)
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glClearColor referenced in function "void __cdecl draw(struct GLFWwindow *)" (?draw@@YAXPEAUGLFWwindow@@@Z)
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glCompileShader referenced in function "unsigned int __cdecl createShader(unsigned int,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createShader@@YAIIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glCompileShader
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glCreateProgram referenced in function "unsigned int __cdecl createProgram(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createProgram@@YAIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@0@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glCreateProgram
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glCreateShader referenced in function "unsigned int __cdecl createShader(unsigned int,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createShader@@YAIIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glCreateShader
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glDeleteProgram referenced in function main
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glDeleteProgram
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glDeleteShader referenced in function "unsigned int __cdecl createProgram(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createProgram@@YAIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@0@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glDeleteShader
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glDeleteTextures referenced in function main
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glDeleteTextures
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glDrawElements referenced in function "void __cdecl draw(struct GLFWwindow *)" (?draw@@YAXPEAUGLFWwindow@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glDrawElements
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glEnableVertexAttribArray referenced in function "void __cdecl draw(struct GLFWwindow *)" (?draw@@YAXPEAUGLFWwindow@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glEnableVertexAttribArray
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glGenTextures referenced in function "unsigned int __cdecl createTextureFromFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createTextureFromFile@@YAIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glGenTextures
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glGetAttribLocation referenced in function main
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glGetAttribLocation
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glGetProgramiv referenced in function "unsigned int __cdecl createProgram(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createProgram@@YAIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@0@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glGetProgramiv
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glGetProgramInfoLog referenced in function "unsigned int __cdecl createProgram(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createProgram@@YAIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@0@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glGetProgramInfoLog
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glGetShaderiv referenced in function "unsigned int __cdecl createShader(unsigned int,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createShader@@YAIIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glGetShaderiv
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glGetShaderInfoLog referenced in function "unsigned int __cdecl createShader(unsigned int,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createShader@@YAIIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glGetShaderInfoLog
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glGetUniformLocation referenced in function main
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glGetUniformLocation
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glLinkProgram referenced in function "unsigned int __cdecl createProgram(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createProgram@@YAIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@0@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glLinkProgram
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glShaderSource referenced in function "unsigned int __cdecl createShader(unsigned int,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createShader@@YAIIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glShaderSource
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glTexImage2D referenced in function "unsigned int __cdecl createTextureFromFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createTextureFromFile@@YAIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glTexImage2D
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glTexParameteri referenced in function "unsigned int __cdecl createTextureFromFile(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createTextureFromFile@@YAIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glTexParameteri
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glUniform1i referenced in function "void __cdecl draw(struct GLFWwindow *)" (?draw@@YAXPEAUGLFWwindow@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glUniform1i
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glUseProgram referenced in function "void __cdecl draw(struct GLFWwindow *)" (?draw@@YAXPEAUGLFWwindow@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glUseProgram
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glValidateProgram referenced in function "unsigned int __cdecl createProgram(class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &)" (?createProgram@@YAIAEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@0@Z)
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glVertexAttribPointer referenced in function "void __cdecl draw(struct GLFWwindow *)" (?draw@@YAXPEAUGLFWwindow@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glVertexAttribPointer
main.cpp.obj : error LNK2019: unresolved external symbol __imp_glViewport referenced in function "void __cdecl draw(struct GLFWwindow *)" (?draw@@YAXPEAUGLFWwindow@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2001: unresolved external symbol __imp_glViewport
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglCreateContext referenced in function _glfwCreateContextEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglCreateWindowSurface referenced in function _glfwCreateContextEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglDestroyContext referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglDestroySurface referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglGetConfigAttrib referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglGetConfigs referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglGetDisplay referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglGetError referenced in function _glfwCreateContextEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglGetProcAddress referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglInitialize referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglMakeCurrent referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglQueryString referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglSwapBuffers referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglTerminate referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglSwapInterval referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglBindAPI referenced in function _glfwCreateContextEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglGetPlatformDisplayEXT referenced in function _glfwInitEGL
glfw3.lib(egl_context.c.obj) : error LNK2019: unresolved external symbol __imp_eglCreatePlatformWindowSurfaceEXT referenced in function _glfwCreateContextEGL
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glBindBuffer referenced in function "bool __cdecl ImGui_ImplOpenGL3_CreateDeviceObjects(void)" (?ImGui_ImplOpenGL3_CreateDeviceObjects@@YA_NXZ)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glBlendEquation referenced in function "void __cdecl ImGui_ImplOpenGL3_SetupRenderState(struct ImDrawData *,int,int,unsigned int)" (?ImGui_ImplOpenGL3_SetupRenderState@@YAXPEAUImDrawData@@HHI@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glBlendEquationSeparate referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glBlendFuncSeparate referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glBufferData referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glBufferSubData referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glDeleteBuffers referenced in function "void __cdecl ImGui_ImplOpenGL3_DestroyDeviceObjects(void)" (?ImGui_ImplOpenGL3_DestroyDeviceObjects@@YAXXZ)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glDetachShader referenced in function "bool __cdecl ImGui_ImplOpenGL3_CreateDeviceObjects(void)" (?ImGui_ImplOpenGL3_CreateDeviceObjects@@YA_NXZ)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glDisable referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glDisableVertexAttribArray referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glEnable referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glGenBuffers referenced in function "bool __cdecl ImGui_ImplOpenGL3_CreateDeviceObjects(void)" (?ImGui_ImplOpenGL3_CreateDeviceObjects@@YA_NXZ)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glGetIntegerv referenced in function "bool __cdecl ImGui_ImplOpenGL3_CreateDeviceObjects(void)" (?ImGui_ImplOpenGL3_CreateDeviceObjects@@YA_NXZ)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glGetVertexAttribiv referenced in function "public: void __cdecl ImGui_ImplOpenGL3_VtxAttribState::GetState(int)" (?GetState@ImGui_ImplOpenGL3_VtxAttribState@@QEAAXH@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glGetVertexAttribPointerv referenced in function "public: void __cdecl ImGui_ImplOpenGL3_VtxAttribState::GetState(int)" (?GetState@ImGui_ImplOpenGL3_VtxAttribState@@QEAAXH@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glIsEnabled referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glScissor referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
imgui.lib(imgui_impl_opengl3.cpp.obj) : error LNK2019: unresolved external symbol __imp_glUniformMatrix4fv referenced in function "void __cdecl ImGui_ImplOpenGL3_SetupRenderState(struct ImDrawData *,int,int,unsigned int)" (?ImGui_ImplOpenGL3_SetupRenderState@@YAXPEAUImDrawData@@HHI@Z)
src\opengles_angle\vcpkg_demo_opengles_angle.exe : fatal error LNK1120: 65 unresolved externals
```

</details>

@xiaozhuai, I follow your steps can't reproduce your error:

1. Install angle
````
   PS F:\Feature-test\port-bug\vcpkg> ./vcpkg install angle:x64-windows-static --editable
Computing installation plan...
The following packages are already installed:
    angle[core]:x64-windows-static -> chromium_5414#2
angle:x64-windows-static is already installed
Restored 0 package(s) from C:\Users\vzhli17\AppData\Local\vcpkg\archives in 167 us. Use --debug to see more details.
Total install time: 1.88 ms
The package angle provides unofficial CMake targets:

    find_package(unofficial-angle REQUIRED CONFIG)
    target_link_libraries(main PRIVATE unofficial::angle::libGLESv2)

    # Or use the EGL target
    find_package(unofficial-angle REQUIRED CONFIG)
    target_link_libraries(main PRIVATE unofficial::angle::libEGL)
````
2. Clone https://github.com/xiaozhuai/vcpkg_demo recursively, and comment out other demos instead of `opengles_angle`
   
![image](https://user-images.githubusercontent.com/63675417/236179838-49c6bb1a-c77b-4e7a-8688-d665a262a14d.png)

3. Build project, and you should see the same error log.
````
PS F:\Feature-test\port-bug\vcpkg_demo\build> cmake .. -G "Visual Studio 17 2022" -A x64  -DVCPKG_TARGET_TRIPLET=x64-windows-static -DCMAKE_BUILD_TYPE=Debug  -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/port-bug/vcpkg/scripts/buildsystems/vcpkg.cmake
-- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
CMake Error at CMakeLists.txt:18 (add_subdirectory):
  The source directory

    F:/Feature-test/port-bug/vcpkg_demo/vendor/glfw

  does not contain a CMakeLists.txt file.


CMake Error at F:/Feature-test/port-bug/vcpkg/scripts/buildsystems/vcpkg.cmake:852 (_find_package):
  By not providing "FindStb.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Stb", but
  CMake did not find one.

  Could not find a package configuration file provided by "Stb" with any of
  the following names:

    StbConfig.cmake
    stb-config.cmake

  Add the installation prefix of "Stb" to CMAKE_PREFIX_PATH or set "Stb_DIR"
  to a directory containing one of the above files.  If "Stb" provides a
  separate development package or SDK, be sure it has been installed.
Call Stack (most recent call first):
  src/opengles_angle/CMakeLists.txt:9 (find_package)

````

This issue could be reproduced with vcpkg 2023-04-07-bedcba5172f5e4b91caac660ab7afe92c27a9895 and VS2022 17.5.4:
````
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2001: unresolved external symbol __imp_glVertexAttribPointer
5>main.obj : error LNK2019: unresolved external symbol __imp_glViewport referenced in function "void __cdecl draw(struct GLFWwindow *)" (?draw@@YAXPEAUGLFWwindow@@@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2001: unresolved external symbol __imp_glViewport
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglCreateContext referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglCreateWindowSurface referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglDestroyContext referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglDestroySurface referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglGetConfigAttrib referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglGetConfigs referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglGetDisplay referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglGetError referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglGetProcAddress referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglInitialize referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglMakeCurrent referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglQueryString referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglSwapBuffers referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglTerminate referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglSwapInterval referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglBindAPI referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglGetPlatformDisplayEXT referenced in function _glfwInitEGL
5>glfw3.lib(egl_context.obj) : error LNK2019: unresolved external symbol __imp_eglCreatePlatformWindowSurfaceEXT referenced in function _glfwInitEGL
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glBindBuffer referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glBlendEquation referenced in function "void __cdecl ImGui_ImplOpenGL3_SetupRenderState(struct ImDrawData *,int,int,unsigned int)" (?ImGui_ImplOpenGL3_SetupRenderState@@YAXPEAUImDrawData@@HHI@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glBlendEquationSeparate referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glBlendFuncSeparate referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glBufferData referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glBufferSubData referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glDeleteBuffers referenced in function "void __cdecl ImGui_ImplOpenGL3_DestroyDeviceObjects(void)" (?ImGui_ImplOpenGL3_DestroyDeviceObjects@@YAXXZ)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glDetachShader referenced in function "bool __cdecl ImGui_ImplOpenGL3_CreateDeviceObjects(void)" (?ImGui_ImplOpenGL3_CreateDeviceObjects@@YA_NXZ)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glDisable referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glDisableVertexAttribArray referenced in function "public: void __cdecl ImGui_ImplOpenGL3_VtxAttribState::SetState(int)" (?SetState@ImGui_ImplOpenGL3_VtxAttribState@@QEAAXH@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glEnable referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glGenBuffers referenced in function "bool __cdecl ImGui_ImplOpenGL3_CreateDeviceObjects(void)" (?ImGui_ImplOpenGL3_CreateDeviceObjects@@YA_NXZ)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glGetIntegerv referenced in function "bool __cdecl ImGui_ImplOpenGL3_Init(char const *)" (?ImGui_ImplOpenGL3_Init@@YA_NPEBD@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glGetVertexAttribiv referenced in function "public: void __cdecl ImGui_ImplOpenGL3_VtxAttribState::GetState(int)" (?GetState@ImGui_ImplOpenGL3_VtxAttribState@@QEAAXH@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glGetVertexAttribPointerv referenced in function "public: void __cdecl ImGui_ImplOpenGL3_VtxAttribState::GetState(int)" (?GetState@ImGui_ImplOpenGL3_VtxAttribState@@QEAAXH@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glIsEnabled referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glScissor referenced in function "void __cdecl ImGui_ImplOpenGL3_RenderDrawData(struct ImDrawData *)" (?ImGui_ImplOpenGL3_RenderDrawData@@YAXPEAUImDrawData@@@Z)
5>imgui.lib(imgui_impl_opengl3.obj) : error LNK2019: unresolved external symbol __imp_glUniformMatrix4fv referenced in function "void __cdecl ImGui_ImplOpenGL3_SetupRenderState(struct ImDrawData *,int,int,unsigned int)" (?ImGui_ImplOpenGL3_SetupRenderState@@YAXPEAUImDrawData@@HHI@Z)
5>F:\Feature-test\port-bug\vcpkg_demo\build\src\opengles_angle\Debug\vcpkg_demo_opengles_angle.exe : fatal error LNK1120: 65 unresolved externals
5>Done building project "vcpkg_demo_opengles_angle.vcxproj" -- FAILED.
6>------ Skipped Build: Project: ALL_BUILD, Configuration: Debug x64 ------
6>Project not selected to build for this solution configuration 
========== Build: 3 succeeded, 1 failed, 0 up-to-date, 2 skipped ==========
========== Build started at 10:43 AM and took 11.541 seconds ==========

````

--- a/ports/angle/cmake-buildsystem/CMakeLists.txt
+++ b/ports/angle/cmake-buildsystem/CMakeLists.txt
@@ -58,6 +58,16 @@ if(ANGLE_USE_D3D11_COMPOSITOR_NATIVE_WINDOW)
     endif()
     set(angle_enable_d3d11_compositor_native_window TRUE)
 endif()
+if(NOT BUILD_SHARED_LIBS)
+    add_definitions(
+        -DANGLE_EXPORT=
+        -DANGLE_STATIC=1
+        -DANGLE_UTIL_EXPORT=
+        -DEGLAPI=
+        -DGL_APICALL=
+        -DGL_API=
+    )
+endif()
 
 find_package(ZLIB REQUIRED)
 
@@ -416,20 +426,52 @@ install(TARGETS ${_installableTargets} EXPORT ANGLEExport
 install(EXPORT ANGLEExport FILE unofficial-angle-targets.cmake NAMESPACE unofficial::angle:: DESTINATION share/unofficial-angle)
 install(FILES unofficial-angle-config.cmake DESTINATION share/unofficial-angle)
 
-if(NOT DISABLE_INSTALL_HEADERS)
-    install(
-        DIRECTORY "${ANGLE_FRAMEWORK_HEADERS_DIR}"
-        DESTINATION include
-        FILES_MATCHING
-        PATTERN "*.h"
-        PATTERN "*.inc"
-        PATTERN "CL" EXCLUDE
-        PATTERN "GLSLANG" EXCLUDE
-        PATTERN "egl.h" EXCLUDE
-        PATTERN "eglext.h" EXCLUDE
-        PATTERN "eglplatform.h" EXCLUDE
-        PATTERN "KHR" EXCLUDE
-        PATTERN "WGL" EXCLUDE
-        PATTERN "export.h" EXCLUDE
-    )
+install(
+    DIRECTORY "${ANGLE_FRAMEWORK_HEADERS_DIR}/"
+    DESTINATION include
+    FILES_MATCHING
+    PATTERN "*.h"
+    PATTERN "*.inc"
+    PATTERN "CL" EXCLUDE
+    PATTERN "GLSLANG" EXCLUDE
+    PATTERN "EGL/egl.h" EXCLUDE
+    PATTERN "EGL/eglext.h" EXCLUDE
+    PATTERN "EGL/eglplatform.h" EXCLUDE
+    PATTERN "KHR" EXCLUDE
+    PATTERN "WGL" EXCLUDE
+    PATTERN "export.h" EXCLUDE
+    PATTERN "GLES/egl.h" EXCLUDE
+    PATTERN "GLES/gl.h" EXCLUDE
+    PATTERN "GLES/glext.h" EXCLUDE
+    PATTERN "GLES/glplatform.h" EXCLUDE
+    PATTERN "GLES2/gl2.h" EXCLUDE
+    PATTERN "GLES2/gl2ext.h" EXCLUDE
+    PATTERN "GLES2/gl2platform.h" EXCLUDE
+    PATTERN "GLES3/gl3.h" EXCLUDE
+    PATTERN "GLES3/gl31.h" EXCLUDE
+    PATTERN "GLES3/gl32.h" EXCLUDE
+    PATTERN "GLES3/gl3platform.h" EXCLUDE
+)
+install(
+    DIRECTORY "${ANGLE_FRAMEWORK_HEADERS_DIR}/"
+    DESTINATION include/angle
+    FILES_MATCHING
+    PATTERN "*.h"
+    PATTERN "*.inc"
+    PATTERN "CL" EXCLUDE
+    PATTERN "GLSLANG" EXCLUDE
+    PATTERN "EGL/egl.h" EXCLUDE
+    PATTERN "EGL/eglext.h" EXCLUDE
+    PATTERN "EGL/eglplatform.h" EXCLUDE
+    PATTERN "KHR" EXCLUDE
+    PATTERN "WGL" EXCLUDE
+    PATTERN "export.h" EXCLUDE
+)
+
+if(NOT BUILD_SHARED_LIBS)
+    foreach(angle_target EGL GLESv2 ANGLE)
+        if(TARGET ${angle_target})
+            target_compile_definitions(${angle_target} INTERFACE $<INSTALL_INTERFACE:KHRONOS_STATIC>)
+        endif()
+    endforeach()
 endif()

--- a/ports/angle/portfile.cmake
+++ b/ports/angle/portfile.cmake
@@ -168,31 +168,19 @@ vcpkg_copy_pdbs()
 
 file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
 
-# File conflict with opengl-registry! Make sure headers are similar on Update!
-# angle defines some additional entrypoints.
-# opengl-registry probably needs an upstream update to account for those
-# Due to that all angle headers get moved to include/angle.
-# If you want to use those instead of the onces provided by opengl-registry make sure
-# VCPKG_INSTALLED_DIR/include/angle is before VCPKG_INSTALLED_DIR/include
-file(GLOB_RECURSE angle_includes "${CURRENT_PACKAGES_DIR}/include")
-file(COPY ${angle_includes} DESTINATION "${CURRENT_PACKAGES_DIR}/include/angle")
-
-set(_double_files
-    "include/GLES/egl.h"
-    "include/GLES/gl.h"
-    "include/GLES/glext.h"
-    "include/GLES/glplatform.h"
-    "include/GLES2/gl2.h"
-    "include/GLES2/gl2ext.h"
-    "include/GLES2/gl2platform.h"
-    "include/GLES3/gl3.h"
-    "include/GLES3/gl31.h"
-    "include/GLES3/gl32.h"
-    "include/GLES3/gl3platform.h")
-foreach(_file ${_double_files})
-    if(EXISTS "${CURRENT_PACKAGES_DIR}/${_file}")
-        file(REMOVE "${CURRENT_PACKAGES_DIR}/${_file}")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+# Remove empty directories inside include directory
+file(GLOB directory_children RELATIVE "${CURRENT_PACKAGES_DIR}/include" "${CURRENT_PACKAGES_DIR}/include/*")
+foreach(directory_child ${directory_children})
+    if(IS_DIRECTORY "${CURRENT_PACKAGES_DIR}/include/${directory_child}")
+        file(GLOB_RECURSE subdirectory_children "${CURRENT_PACKAGES_DIR}/include/${directory_child}/*")
+        if("${subdirectory_children}" STREQUAL "")
+            file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/include/${directory_child}")
+        endif()
     endif()
 endforeach()
+unset(subdirectory_children)
+unset(directory_child)
+unset(directory_children)
 
 file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")

--- a/ports/angle/vcpkg.json
+++ b/ports/angle/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "angle",
   "version-string": "chromium_5414",
-  "port-version": 2,
+  "port-version": 3,
   "description": [
     "A conformant OpenGL ES implementation for Windows, Mac and Linux.",
     "The goal of ANGLE is to allow users of multiple operating systems to seamlessly run WebGL and other OpenGL ES content by translating OpenGL ES API calls to one of the hardware-supported APIs available for that platform. ANGLE currently provides translation from OpenGL ES 2.0 and 3.0 to desktop OpenGL, OpenGL ES, Direct3D 9, and Direct3D 11. Support for translation from OpenGL ES to Vulkan is underway, and future plans include compute shader support (ES 3.1) and MacOS support."

--- a/versions/a-/angle.json
+++ b/versions/a-/angle.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "d5671959f9f2a59ecd232c72df54682bb10a14be",
+      "version-string": "chromium_5414",
+      "port-version": 3
+    },
+    {
       "git-tree": "784aa16c1dacc9aedb49de3bd1393bb6ef6b853d",
       "version-string": "chromium_5414",
       "port-version": 2

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -122,7 +122,7 @@
     },
     "angle": {
       "baseline": "chromium_5414",
-      "port-version": 2
+      "port-version": 3
     },
     "annoy": {
       "baseline": "1.17.2",

